src/Core/Checkout/Cart/CartRuleLoader.php

```diff
@@ -116,6 +116,7 @@ private function load(SalesChannelContext $context, Cart $cart, CartBehavior $be
 
                 // update matching rules in context
                 $context->setRuleIds($rules->getIds());
+                $context->setAreaRuleIds($rules->getIdsByArea());
```

Comment:
can you add a small unit test?

Reply:
Will do that :-) I also noticed that I missed that for the initial cart processing the area rule ids are also missing. I will update the PR accordingly.